source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -2402,19 +2453,27 @@ void COpenGLDriver::setRenderStates3DMode()
 		ResetRenderStates = true;
 	}
 
+    //actually i'm commenting-out ALMOST all in this if-statemenet
 	if (ResetRenderStates || LastMaterial != Material)
 	{
 		// unset old material
 
-		if (LastMaterial.MaterialType != Material.MaterialType &&
+		/* TODO what OnUnsetMaterial() is responsible for? Can i remove this?
+        if (LastMaterial.Pipeline != Material.Pipeline &&
 				static_cast<uint32_t>(LastMaterial.MaterialType) < MaterialRenderers.size())
-			MaterialRenderers[LastMaterial.MaterialType].Renderer->OnUnsetMaterial();
+			MaterialRenderers[LastMaterial.MaterialType].Renderer->OnUnsetMaterial();*/
 
 		// set new material.
-		if (static_cast<uint32_t>(Material.MaterialType) < MaterialRenderers.size())
-			MaterialRenderers[Material.MaterialType].Renderer->OnSetMaterial(
+		//if (static_cast<uint32_t>(Material.MaterialType) < MaterialRenderers.size())
+			MaterialRenderers[0].Renderer->OnSetMaterial(
```

Comment:
did this do anything else other than call the callback?

Reply:
from COpenGLSLmaterialRenderer:
```
    Driver->setBasicRenderStates(material, lastMaterial, resetAllRenderstates);

	if (material.Pipeline != lastMaterial.Pipeline || resetAllRenderstates)
	{
		if (BaseMaterial==EMT_TRANSPARENT_ADD_COLOR)
			glBlendFunc(GL_SRC_ALPHA, GL_ONE);
        else if (BaseMaterial==EMT_TRANSPARENT_ALPHA_CHANNEL||BaseMaterial==EMT_TRANSPARENT_VERTEX_ALPHA)
			glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
	}
```